channeldb/graph_cache.go

```diff
@@ -48,11 +134,27 @@ type DirectedChannel struct {
 	// Capacity is the announced capacity of this channel in satoshis.
 	Capacity btcutil.Amount
 
-	// OutPolicy is the outgoing policy from this node *to* the other node.
-	OutPolicy *ChannelEdgePolicy
+	// OutPolicySet is a boolean that indicates whether the node has an
+	// outgoing policy set. For pathfinding only the existence of the policy
+	// is important to know, not the actual content.
+	OutPolicySet bool
 
 	// InPolicy is the incoming policy *from* the other node to this node.
-	InPolicy *ChannelEdgePolicy
+	InPolicy *CachedEdgePolicy
+}
+
+// DeepCopy creates a deep copy of the channel, including the incoming policy.
+func (c *DirectedChannel) DeepCopy() *DirectedChannel {
+	channelCopy := *c
+
+	if channelCopy.InPolicy != nil {
+		inPolicyCopy := *channelCopy.InPolicy
+		channelCopy.InPolicy = &inPolicyCopy
+		channelCopy.InPolicy.ToNodePubKey = nil
```

Comment:
nit: maybe add some comments explaining why deep copy sets the fields to nil? Reading from below seems like they are overwritten by the path finding algorithm.